### DISCUSSION TITLE
How can I import with settings in Editor mode?

### DISCUSSION THREAD

**Discussion by from2001** [atteneder/glTFast#682] [Q&A]
I'd like to import glTF/glb 3D models with specific import settings.

The usage of settings of runtime import is described in [this page](https://github.com/atteneder/glTFast/blob/openupm/Documentation~/ImportRuntime.md), but settings for editor import is not explained.

I wrote my own materialGenerator, and it works at runtime. How can I import with settings in Editor mode? I want to use the materialGenerator for editor import.

Best,

**Comment by atteneder** [comment]
Hi,

I love the idea!

From an UX perspective you'd have a field in the Importer's inspector that you can assign a script (that implements `IMaterialGenerator`) to.

Technically this is tricky, as you cannot serialize a reference to a script, afaik. It might work if the serialized field is a string (type name) and the Importer converts it to the actual class via reflection.

That would require some more research. Anyone is invited to play with the idea and submit a draft!
